### ISSUE TITLE
Class Siren foward coords requires grad?

### ISSUE THREAD

**Issue by hpicsk** [vsitzmann/siren#54]
In Colab notebook implementation of Class Siren foward
`    def forward(self, coords):
        coords = coords.clone().detach().requires_grad_(True) # allows to take derivative w.r.t. input
        output = self.net(coords)
        return output, coords   `

why are coords set require_grad_(True)? Shouldn't the weights of the network only updated, not coordinates?
